nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh

```diff
@@ -100,80 +161,61 @@ public:
                                           taken for training model */
   unsigned int num_validation_samples; /**< The number of validation samples to
                                           be taken for validation model */
-  unsigned int total_num_samples;      /**< Total number of samples received for
-                                          creating model */
-  unsigned int num_epochs;             /**< The number of epoch */
-  unsigned int num_push_data;          /**< The number of samples pushed by
-                                          NNStreamer(tensor_trainer) */
-  std::string model_config;
-  std::string model_save_path; /**< Model is finally stored */
-  std::string model_load_path; /**< Path to load an exisiting model to use for
-                                  training a new model */
-
-  ml::train::RunStats train_stats;
-  ml::train::RunStats valid_stats;
-
+  unsigned int tensors_inputsize[NNS_TENSOR_SIZE_LIMIT];
+  unsigned int
+    num_tensors; /**< The number of tensors in the received a sample */
   GstTensorTrainerEventNotifier *notifier; /**< a handle of event notify */
-  bool stop_model_training;
-
-private:
   std::unique_ptr<ml::train::Model> model;
 };
 
 /**
- * @brief Manage input tensors data and information
+ * @brief Stores input tensors to the queue
  */
-class InputTensorsInfo {
+class TensorsQueue {
 public:
   /**
-   * @brief Construct a new InputTensorsInfo object
-   * @param _total_num_samples Total number of samples received for creating
+   * @brief Construct a new TensorsQueue object
+   * @param _num_of_samples Total number of samples received for creating
    * model
-   * @param _num_inputs The number of tensors used as input in the received a
+   * @param _num_of_inputs The number of tensors used as input in the received a
    * sample
-   * @param _num_labels The number of tensors used as label in the received a
+   * @param _num_of_labels The number of tensors used as label in the received a
    * sample
    * @param _tensors_size[] size of each tensor in a sample
    */
-  InputTensorsInfo(unsigned int _total_num_samples, unsigned int _num_inputs,
-                   unsigned int _num_labels, unsigned int _tensors_size[]);
+  TensorsQueue(unsigned int _num_of_samples, unsigned int _num_of_inputs,
```

Comment:
It looks the other four constructors are required.

Reply:
Thank you.